### PR TITLE
API declarations

### PR DESCRIPTION
**Pull request by igorkofman**
This PR addresses issue https://github.com/covid-projections/covid-data-model/issues/< >

### Please Check
- [ ] Will the *current* schema in [api docs](https://github.com/covid-projections/covid-data-model/blob/master/api/README.md) be updated with this PR?
- [ ] Are tests passing?

### CODE REVIEW

File: api/can_api_definition.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+"""
+Documented https://www.dropbox.com/scl/fi/o4bec2kz8dkcxdtabtqda/CAN-V1-API-Draft.paper?dl=0&rlkey=f7elx3zmo6tt5s7mj5nvap9a0
+
+"""
+
+class _ResourceUsageProjection(pydantic.BaseModel):
+    peakShortfall: int = pydantic.Field(
+        ..., description="Shortfall of resource needed at the peek utilization"
+    )
```
Comment (goldblatt) [inline api/can_api_definition.py:13]: this needs to be optional as it might be none

File: api/can_api_definition.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    peakDate: datetime.date = pydantic.Field(
+        ..., description="Date of peak resource utilization"
+    )
```
Comment (goldblatt) [inline api/can_api_definition.py:16]: this also needs to be optional

File: api/can_api_definition.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    shortageStartDate: Optional[datetime.date] = pydantic.Field(
+        ..., description="Date when resource shortage begins"
+    )
+
+class _Projections(pydantic.BaseModel):
```
Comment (goldblatt) [inline api/can_api_definition.py:21]: also think specififying the projection window would be helpful

File: libs/functions/generate_api.py [hunk 4]
```diff
@@ -22,20 +22,27 @@ def generate_api_for_projection(projection):
...
-        county_result = CANPredictionAPIRow(
+        county_result = CovidActNowCountySummary(
+            lat=county_row[rc.LATITUDE],
+            long=county_row[rc.LONGITUDE],
+            actuals=None,
             stateName=county_row[rc.STATE],
             countyName=county_row[rc.COUNTY],
             fips=county_row[rc.FIPS],
-            lastUpdatedDate=county_row[rc.LAST_UPDATED],
+            lastUpdatedDate=_get_date_or_none(county_row[rc.LAST_UPDATED]),
```
Comment (goldblatt) [inline libs/functions/generate_api.py:44]: can this be empty?

File: api/can_api_definition.py (hunk 1 above)
Comment (igorkofman) [reply api/can_api_definition.py:13]: alternatively it could be 0?

File: api/can_api_definition.py (hunk 2 above)
Comment (igorkofman) [reply api/can_api_definition.py:16]: agreed

File: libs/functions/generate_api.py (hunk 4 above)
Comment (igorkofman) [reply libs/functions/generate_api.py:44]: it shouldn't be

File: api/can_api_definition.py (hunk 1 above)
Comment (goldblatt) [reply api/can_api_definition.py:13]: i think we pass in a <null> for this so we could either make it 0 elsewhere or make it none

**Pull request merged**